lib/matplotlib/colorbar.py

```diff
@@ -573,6 +574,13 @@ def set_ticks(self, ticks, update_ticks=True):
 
     def get_ticks(self, minor=False):
         """Return the x ticks as a list of locations"""
+        if self._tick_data_values is None:
```

Comment:
Not quite following the logic here. We populate `self._tick_data_values` via `update_ticks() -> self._ticker` but not if `_use_auto_colorbar_locator`. It seems that if we use `self._tick_data_values` to store the ticks of a fixed locator. However, this mechanism is quite hidden and seems intransparent.

Would it make sense, to either rename `_tick_data_values` to something like `_fixed_tick_data_values` or alternatively get rid of that local variable and obtain the value dynamically?

Reply:
@timhoffm fixed.

We cannot (trivially) get the value automatically because in the old days, all colorbars went from 0-1, and then were labeled from vmin to vmax by hand.  Some types of colorbars are still made that way, so `long_axis.get_majorticklocs()` returns a list of floats between 0 and 1.  

I think a todo is to go back and make those old-style colorbars the same as the new style, so they are *all* done in data space, and then we can make this homogeneous.  But for a point release, this gets things working again...

Reply:
Anyway, added a comment, and changed the name of the private (I hope?) variable...